lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php

```diff
@@ -255,11 +255,20 @@ public function loadMetadataForClass($className, ClassMetadata $metadata)
 
         if (isset($xmlRoot->embedded)) {
             foreach ($xmlRoot->embedded as $embeddedMapping) {
+                $columnPrefix = isset($embeddedMapping['column-prefix'])
```

Comment:
> The only difference compared to `YML`, etc... is that any incoming value from a XML attribute will be always string :) 

That's indeed why you need an additional attribute.